integration-tests/tests/src/hooks/open-realm-before.ts

```diff
@@ -29,23 +29,32 @@ export function openRealmBefore(config: LocalConfiguration | SyncedConfiguration
       // Upload the schema, ensuring a valid connection
       await this.realm.syncSession.uploadAllLocalChanges();
     }
-  });
+  }
+}
 
-  // Clean up afterwards
-  after(function (this: RealmContext) {
-    if (this.realm) {
-      this.realm.close();
-      delete this.realm;
-    } else {
-      throw new Error("Expected a 'realm' in the context");
-    }
-    if (this.config) {
-      Realm.deleteFile(this.config);
-      delete this.config;
-    } else {
-      throw new Error("Expected a 'config' in the context");
-    }
-    // Clearing the test state to ensure the sync session gets completely reset and nothing is cached between tests
-    Realm.clearTestState();
-  });
-}
\ No newline at end of file
+export function closeRealm(this: RealmContext) {
+  if (this.realm) {
+    this.realm.close();
+    delete this.realm;
+  } else {
+    throw new Error("Expected a 'realm' in the context");
+  }
+  if (this.config) {
+    Realm.deleteFile(this.config);
+    delete this.config;
+  } else {
+    throw new Error("Expected a 'config' in the context");
+  }
+  // Clearing the test state to ensure the sync session gets completely reset and nothing is cached between tests
+  Realm.clearTestState();
+}
+
+export function openRealmBeforeEach(config: LocalConfiguration | SyncedConfiguration = {}) {
```

Comment:
I did this refactor to make it easier to run this hook either `beforeEach` or `before`, but it turned out I didn't need it anyway.